tests/unit/test_state.py

```diff
@@ -129,3 +129,29 @@ def test_proxyconfig_from_charm_env(
     assert config.http_proxy == proxy_config.http_proxy
     assert config.https_proxy == proxy_config.https_proxy
     assert config.no_proxy == proxy_config.no_proxy
+
+
+def test_plugins_config_none(harness: Harness):
```

Comment:
You should try to avoid using harness to test any piece of code different from the charm itself and the libraries. Perhaps you can define and instantiate a very simple charm

Reply:
Since the charm (`ops.CharmBase`) requires a `ops.Framework`, I think a harness would need to be injected anyways. Would there be an approach that I'm not seeing that's simpler?
This is the [example](https://github.com/canonical/loki-k8s-operator/blob/0daf783317b32fe31f6a4500ff01f98c406aed49/tests/unit/test_consumer.py#L18) I found in the loki-k8s codebase.

Otherwise, I could mock the framework which would involve a bit of defining the return types.
```
class TestingCharm(ops.CharmBase):
        """A simple charm for testing."""

    mock_framework = MagicMock(spec=ops.Framework)
    # mock_framework requires defining some return values to avoid errors like the following:
    # Invalid time range <MagicMock name='mock.model.config.get()' id='139838944989616'>, time range must be 
    # an integer.
    config = state.State.from_charm(TestingCharm(mock_framework))
```

Reply:
How about https://github.com/canonical/saml-integrator-operator/blob/main/tests/unit/test_charm_state.py#L33?

Reply:
Oof why did I think of mocking the framework but not the charm? 😵‍💫 Updated, thank you!